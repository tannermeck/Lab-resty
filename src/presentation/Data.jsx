import React from 'react';
import ReactJson from 'react-json-view';
import './data.css';

export const Data = ({data}) => {
    return (
        <section>
            <ReactJson src={data} />
        </section>
    )
}