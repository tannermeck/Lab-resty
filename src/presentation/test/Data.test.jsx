import React from 'react';
import ReactJson from 'react-json-view';
import { Data } from '../Data.jsx';
import { render } from '@testing-library/react';

describe('should render a data component', () => {
    it('it should render an object with reactjson', () => {
        const { asFragment } = render(
        <Data data={[{"Hello": "This is a test"}]} />);
        expect(asFragment()).toMatchSnapshot()
    })
})