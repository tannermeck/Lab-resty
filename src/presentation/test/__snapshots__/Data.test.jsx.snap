// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a data component it should render an object with reactjson 1`] = `
<DocumentFragment>
  <section>
    <div
      class="react-json-view"
      style="font-family: monospace; cursor: default; background-color: rgba(0, 0, 0, 0); position: relative;"
    >
      <div
        class="pretty-json-container object-container"
      >
        <div
          class="object-content"
        >
          <div
            class="object-key-val"
          >
            <span>
              <span
                style="display: inline-block; cursor: pointer;"
              >
                <div
                  class="icon-container"
                  style="display: inline-block; width: 17px;"
                >
                  <span
                    class="expanded-icon"
                  >
                    <svg
                      fill="currentColor"
                      style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                      viewBox="0 0 15 15"
                    >
                      <path
                        d="M0 5l6 6 6-6z"
                      />
                    </svg>
                  </span>
                </div>
                <span
                  style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                >
                  <span
                    class="object-key"
                  >
                    <span
                      style="vertical-align: top;"
                    >
                      "
                    </span>
                    <span>
                      root
                    </span>
                    <span
                      style="vertical-align: top;"
                    >
                      "
                    </span>
                  </span>
                  <span
                    style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                  >
                    :
                  </span>
                </span>
                <span
                  style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                >
                  [
                </span>
              </span>
              <div
                class="object-meta-data"
                style="display: inline-block; padding: 0px 0px 0px 10px;"
              >
                <span
                  class="object-size"
                  style="color: rgba(0, 0, 0, 0.3); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                >
                  1 item
                </span>
                <span
                  class="copy-to-clipboard-container"
                  style="vertical-align: top; display: none;"
                  title="Copy to clipboard"
                >
                  <span
                    style="cursor: pointer; display: inline;"
                  >
                    <span
                      class="copy-icon"
                    >
                      <svg
                        fill="currentColor"
                        preserveAspectRatio="xMidYMid meet"
                        style="vertical-align: top; color: rgb(38, 139, 210); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                        viewBox="0 0 40 40"
                      >
                        <g>
                          <path
                            d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                          />
                        </g>
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
            </span>
            <div
              class="pushed-content object-container"
            >
              <div
                class="object-content"
                style="margin-left: 6px;"
              >
                <div
                  class="object-key-val"
                  style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                >
                  <span>
                    <span
                      style="display: inline-block; cursor: pointer;"
                    >
                      <div
                        class="icon-container"
                        style="display: inline-block; width: 17px;"
                      >
                        <span
                          class="expanded-icon"
                        >
                          <svg
                            fill="currentColor"
                            style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                            viewBox="0 0 15 15"
                          >
                            <path
                              d="M0 5l6 6 6-6z"
                            />
                          </svg>
                        </span>
                      </div>
                      <span
                        style="display: inline-block; color: rgb(108, 113, 196); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                      >
                        <span
                          class="array-key"
                        >
                          0
                        </span>
                        <span
                          style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                        >
                          :
                        </span>
                      </span>
                      <span
                        style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                      >
                        {
                      </span>
                    </span>
                    <div
                      class="object-meta-data"
                      style="display: inline-block; padding: 0px 0px 0px 10px;"
                    >
                      <span
                        class="object-size"
                        style="color: rgba(0, 0, 0, 0.3); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                      >
                        1 item
                      </span>
                      <span
                        class="copy-to-clipboard-container"
                        style="vertical-align: top; display: none;"
                        title="Copy to clipboard"
                      >
                        <span
                          style="cursor: pointer; display: inline;"
                        >
                          <span
                            class="copy-icon"
                          >
                            <svg
                              fill="currentColor"
                              preserveAspectRatio="xMidYMid meet"
                              style="vertical-align: top; color: rgb(38, 139, 210); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                              viewBox="0 0 40 40"
                            >
                              <g>
                                <path
                                  d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                />
                              </g>
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </span>
                  <div
                    class="pushed-content object-container"
                  >
                    <div
                      class="object-content"
                      style="margin-left: 6px;"
                    >
                      <div
                        class="variable-row"
                        style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                      >
                        <span>
                          <span
                            class="object-key"
                            style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                          >
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                            <span
                              style="display: inline-block;"
                            >
                              Hello
                            </span>
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                          </span>
                          <span
                            style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                          >
                            :
                          </span>
                        </span>
                        <div
                          class="variable-value"
                          style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                        >
                          <div
                            style="display: inline-block; color: rgb(203, 75, 22);"
                          >
                            <span
                              class="data-type-label"
                              style="font-size: 11px; margin-right: 4px; opacity: 0.8;"
                            >
                              string
                            </span>
                            <span
                              class="string-value"
                              style="cursor: default;"
                            >
                              "This is a test"
                            </span>
                          </div>
                        </div>
                        <span
                          class="copy-to-clipboard-container"
                          style="vertical-align: top; display: none;"
                          title="Copy to clipboard"
                        >
                          <span
                            style="cursor: pointer; display: inline;"
                          >
                            <span
                              class="copy-icon"
                            >
                              <svg
                                fill="currentColor"
                                preserveAspectRatio="xMidYMid meet"
                                style="vertical-align: top; color: rgb(38, 139, 210); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                                viewBox="0 0 40 40"
                              >
                                <g>
                                  <path
                                    d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                  />
                                </g>
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <span
                    class="brace-row"
                  >
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                    >
                      }
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <span
              class="brace-row"
            >
              <span
                style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
              >
                ]
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
